kge/job/job_factory.py

```diff
@@ -0,0 +1,41 @@
+from kge import Config, Dataset
+
+from typing import Optional
+
+
+class JobFactory:
+    @staticmethod
+    def pick_job(job_type=None, config=None):
+        from kge.job import TrainingJob, EvaluationJob, SearchJob
+
+        if job_type is None and config is None:
+            job_type = "train"
+        elif job_type is None:
+            job_type = config.get("job.type")
+
+        if job_type == "train":
+            return TrainingJob
+        elif job_type == "search":
+            return SearchJob
+        elif job_type == "eval":
+            return EvaluationJob
+        else:
+            raise ValueError("unknown job type")
+
+    @staticmethod
+    def create(config: Config, dataset: Dataset, parent_job=None, init=True):
+        job_class = JobFactory.pick_job(config=config)
+        return job_class.create(config, dataset, parent_job=parent_job, init=init)
+
+    @staticmethod
+    def load_from(
+        checkpoint_file: str,
+        job_type=None,
+        config: Optional[Config] = None,
```

Comment:
What's the semantic of this config? Does it override the one from the checkpoint? If so, that would be a good design, imho. We can then, for example, override the job type with a new config that only holds "job.type=x". The job_type argument is then also not needed (one may also use EvalJob.load_from directly).

Reply:
can be used without a config now.
If we call EvalJob.load_from() directly without any config, it will currently not know what type of job to create